### ISSUE TITLE
[Request] Option to remove/hide News and Interests from Taskbar in Windows 10 and disable Game Bar in the Settings.

### ISSUE THREAD

**Issue by R4vz** [cschneegans/unattend-generator#144]
Title. Hoping that you introduce the options mentioned above. 

Also it would be great that the Game Bar is checked automatically while removing Xbox Apps from the Bloatware list. Thank you.

**Comment by wwqwq2313** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modified the Xbox Apps entry in Bloatware.json to separate Game Bar from other Xbox Apps. This allows users to remove Xbox Apps while keeping the Game Bar if desired.

You can review changes in this commit: https://github.com/wwqwq2313/cschneegans-unattend-generator-144/commit/673f074e5cab449f8ce14050217892210d38a41c.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by cschneegans** [comment]
Just looked into this briefly – and the configuration seems to be a total mess. Some versions of Windows 10 use the `ShellFeedsTaskbarViewMode` value in the `[HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Feeds]` key, while others use the `EnShellFeedsTaskbarViewMode` value in the same key.

[tenforums.com](https://www.tenforums.com/tutorials/188597-add-remove-news-interests-icon-taskbar-windows-10-a.html#option3) offers a fine tutorial as usual, but the given numeric values (4b078520, 4e1e580d, 1649fabb) do not match those on my 22H2 machine (7004fee3, 71212960, 6208c68e). One user says: “Well, this isn't all correct, the Data Values are machine specific apparently.”

There is also a [mydigitallife.net](https://forums.mydigitallife.net/threads/taskbarda-widgets-registry-change-is-now-blocked.88547/) thread on this topic. Their PowerShell script suggests that the correct `EnShellFeedsTaskbarViewMode` values are indeed machine-specific and somehow derived from the `MachineId` GUID in the `[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\SQMClient]` registry key.

Under these circumstances I am afraid I cannot add the setting you requested.

**Comment by R4vz** [comment]
Oh my!! Seems like it's much more complicated than I think. Thank you for your response and the explanation.

What about a separate option to disable Game Bar? Like I said in the previous post can you add a function such that removing Xbox Apps will automatically check the option to disable Game Bar.

**Comment by cschneegans** [comment]
> What about a separate option to disable Game Bar? Like I said in the previous post can you add a function such that removing Xbox Apps will automatically check the option to disable Game Bar.

Bear with me. I am not a gamer. I think I have never used the Game Bar, and I probably would not notice if it behaves differently.

There is a setting _Allow your controller to open Game Bar_

![image](https://github.com/user-attachments/assets/455e6785-2829-4c96-b74e-c7437d0ff69a)

which is linked to this registry value:

```
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Microsoft\GameBar]
"UseNexusForGameBarEnabled"=…
```

Am I correct to assume that this setting should be disabled (`=dword:00000000`) automatically when _Xbox Apps_ is selected in the _Remove bloatware_ section?

![image](https://github.com/user-attachments/assets/3b1659d0-649e-437d-92a4-28c25b2f39c5)

**Comment by R4vz** [comment]
> Bear with me. I am not a gamer.I think I have never used the Game Bar, and I probably would not notice if it behaves differently.
I do not game that often too lol and neither have I used Game Bar until now. I've always kept if off since I don't find it any helpful and it's basically useless, especially when you remove all the Xbox Apps.

> There is a setting Allow your controller to open Game Bar
Yes there is. But the thing is, I always have to go to the Settings and disable it manually for every account on my PC. 

Also, I think the screenshot you've shared is of Windows 11. But it's different on latest Windows 10 22H2 as shown below.
![Capture](https://github.com/user-attachments/assets/5952c416-2c7e-4021-a285-c596227fc6ab)

I found this [post](https://www.tenforums.com/tutorials/8637-turn-off-xbox-game-bar-windows-10-a.html#option2) from TenForums which has a different registry key from what you've mentioned. Hopefully it's going to be helpful.

> Am I correct to assume that this setting should be disabled (=dword:00000000) automatically when Xbox Apps is selected in the Remove bloatware section?
Exactly, with an option to opt out if the user wants to just like this.
![Capture11](https://github.com/user-attachments/assets/72c92f6a-7c96-47f0-b46a-4421fda44c4f)

**Comment by cschneegans** [comment]
The actual Game Bar app (which is launched by pressing Win+G) is provided by the `Microsoft.XboxGamingOverlay` package. This package is already removed when the _Xbox Apps_ checkbox in the _Remove bloatware_ section is selected. Therefore, a secondary _Disable Game Bar_ checkbox is unnecessary.

• In Windows 11, is Win+G shortcut is simply ignored when the Game Bar app is uninstalled.

• In Windows 10, however, an annoying pop-up window is shown:

![image](https://github.com/user-attachments/assets/f6349d91-44b3-4032-92c1-0e522d7a98b5)

I assume this pop-up is what bothers you? The _AppCaptureEnabled_ registry value is now set (via https://github.com/cschneegans/unattend-generator/commit/63897c0a0090ae84983ed326bc033e7deff6d27d) to suppress this pop-up.

**Comment by R4vz** [comment]
> The actual Game Bar app (which is launched by pressing Win+G) is provided by the Microsoft.XboxGamingOverlay package. This package is already removed when the Xbox Apps checkbox in the Remove bloatware section is selected. Therefore, a secondary Disable Game Bar checkbox is unnecessary.

> In Windows 10, however, an annoying pop-up window is shown

> I assume this pop-up is what bothers you? The AppCaptureEnabled registry value is now set (via https://github.com/cschneegans/unattend-generator/commit/63897c0a0090ae84983ed326bc033e7deff6d27d) to suppress this pop-up.

Yes sir, the package is removed when I checked in Powershell. And that pop-up is so annoying, specially when you've got custom keyboard shortcuts for certain apps. Hence, the toggle had to be disabled.

Thanks a lot for implementing my request. You sir are great.

**Comment by Hammerfest** [comment]
@cschneegans whats the easiest way to undo this? I do want the xbox apps removed, but several games rely on the gamebar backend to do things like set the cores properly for the X3D multi ccd series of AMD processors

There are some other things broken as well with its removal

**Comment by cschneegans** [comment]
It is my understanding that the Xbox apps _are_ the Gambar's backend?!

In any case, you should be able to reinstall removed packages from the Microsoft Store. Here are the Store links/product IDs for the packages that are removed on Windows 11 24H2 when you select the “Xbox Apps” bloatware removal;

• Microsoft.Xbox.TCUI
`ms-windows-store://pdp/?productid=9NKNC0LD5NN6`

• Microsoft.XboxGamingOverlay
`ms-windows-store://pdp/?productid=9NZKPSTSNW4P`

• Microsoft.XboxIdentityProvider
`ms-windows-store://pdp/?productid=9WZDNCRD1HKW`

• Microsoft.XboxSpeechToTextOverlay
`ms-windows-store://pdp/?productid=9P086NHDNB9W`

**Comment by Hammerfest** [comment]
I think its just the overlay, but I ran out of time for testing, I ran into another issue with things like right click classic menu not applying for newly created users, will ticket that later.

TY, sorry for the weekend ping, never rush for me :D
